sensing/position_estimation.c

```diff
@@ -254,16 +254,25 @@ static void position_estimation_position_correction(position_estimation_t *pos_e
 		}
 		gps_gain = 0.0f;
 	}
-	
+
+	// static float last_sonar = 0.0f;
+	// const  float clip_threshold = 0.9f;
+
+	// if (pos_est->sonar->current_distance - last_sonar < clip_threshold)
+	// {
+	// 	last_sonar = pos_est->sonar->current_distance;
+	// }
+
```

Comment:
I think you mess up here...
if the diff is less than clip_threshold then we use the new value, 
else we use previous sonar value